compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDelegatedPropertyChecker.kt

```diff
@@ -0,0 +1,113 @@
+
+            override fun visitFunctionCall(functionCall: FirFunctionCall) {
+                val errorNamedReference = functionCall.calleeReference as? FirErrorNamedReference ?: return
+                if (errorNamedReference.source?.kind != FirFakeSourceElementKind.DelegatedPropertyAccessor) return
+                val expectedFunctionSignature =
+                    (if (isGet) "getValue" else "setValue") + "(${functionCall.arguments.joinToString(", ") { it.typeRef.coneType.render() }})"
+                val delegateDescription = if (isGet) "delegate" else "delegate for var (read-write property)"
+                fun reportInapplicableDiagnostics(
+                    candidateApplicability: CandidateApplicability,
+                    candidates: Collection<AbstractFirBasedSymbol<*>>
+                ) {
+                    if (candidateApplicability == CandidateApplicability.INAPPLICABLE_WRONG_RECEIVER) {
+                        reporter.reportOn(
+                            errorNamedReference.source,
+                            FirErrors.DELEGATE_SPECIAL_FUNCTION_MISSING,
+                            expectedFunctionSignature,
+                            delegateType,
+                            delegateDescription,
+                            context
+                        )
+                    } else {
+                        reporter.reportOn(
+                            errorNamedReference.source,
+                            FirErrors.DELEGATE_SPECIAL_FUNCTION_NONE_APPLICABLE,
+                            expectedFunctionSignature,
+                            candidates,
+                            context
+                        )
+                    }
+                }
+                when (val diagnostic = errorNamedReference.diagnostic) {
+                    is ConeUnresolvedNameError -> {
+                        reporter.reportOn(
+                            errorNamedReference.source,
+                            FirErrors.DELEGATE_SPECIAL_FUNCTION_MISSING,
+                            expectedFunctionSignature,
+                            delegateType,
+                            delegateDescription,
+                            context
+                        )
+                    }
+                    is ConeAmbiguityError -> {
+                        if (diagnostic.applicability.isSuccess) {
+                            // Match is successful but there are too many matches! So we report DELEGATE_SPECIAL_FUNCTION_AMBIGUITY.
+                            reporter.reportOn(
+                                errorNamedReference.source,
+                                FirErrors.DELEGATE_SPECIAL_FUNCTION_AMBIGUITY,
+                                expectedFunctionSignature,
+                                diagnostic.candidates.map { it.symbol },
+                                context
+                            )
+                        } else {
```

Comment:
What is the case for this else?

Reply:
If

Reply:
If no candidate are applicable (`applicability.isSuccess == false`), then we should report either missing or none applicable.